### ISSUE TITLE
"Not a valid Win32 Application" Windows XP

### ISSUE THREAD

**Issue by kurisubrooks** [atom/electron#3370]
We get the following error when trying to execute a binary. We also tried running it from console, getting the same error. @kaorioka

![pasted image at 2015_11_09 01_11 am](https://cloud.githubusercontent.com/assets/6291467/11020788/4d960aa2-867f-11e5-90df-3f8ca0c148f9.png)

**Comment by magicae** [comment]
XP is not support.

**Comment by paulcbetts** [comment]
Electron only runs on Windows 7 or higher

**Comment by gbaumgart** [comment]
Is there at least a way to compile electron for windows xp ? Chrome runs just fine on windows xp like many other apps, for good reason. On the other side, we'd have to tell people to throw their working hardware in the trash (because in reality you can't update to windows-7 because of missing driver support)

any chance to get this working ? I am willing to spend some time on this. Would be sad if we tend supporting wasteful software only. Our new generations gonna pay a huge price for this things, assuming there is still something left after us.
g

**Comment by paulcbetts** [comment]
@gbaumgart It is a significant amount of work to support Windows XP and we aren't doing to take any PR to try to do it. If you want to try to do it yourself you are of course welcome to.

**Comment by gbaumgart** [comment]
@paulcbetts gonna try that. Or is electron using window platform functions which are only present in windows 7+ ? That would explain "significant work". For now I assume only compiler fiddling and maybe kicking out things not really needed. I guess there are also some path issues, at worst. Still, it worth the work.

**Comment by paulcbetts** [comment]
> Or is electron using window platform functions which are only present in windows 7+ ? 

Yep, this is the case

**Comment by gbaumgart** [comment]
ok, thanks, too bad, but i guess it can't be that much, at worst we switch back to CEF as base.

**Comment by mschnee** [comment]
You can absolutely fork electron and make the chrome parts of it compatible for XP - those are things like the Touch APIs which XP doesn't support. A number of people and organizations have done it already.  Be warned that CEF is also dropping support for XP (it's really chrome dropping support)

However, supporting XP users comes with very significant security considerations, and there is a great deal of engineering overhead in doing so. 

TLDR it's OSS you can do what you want, but should you?

**Comment by gbaumgart** [comment]
@mschnee thanks a lot for the insights. At the end, I want to make our software accessible to people who can't afford windows 7+ only.

I pretty much guess that I don't need to tell you:  I against telling one people to upgrade to something worst when the actual hardware and system is still working well. And security isn't always an issue, my 3k laptop from Sony (2006) is doing just fine, with the the same xp installation and never had any security problem or virus, because you know: there are also people who don't install crap on their beloved machines and take care. On the contrary: do you image how much trash we'd have more if everybody would follow the wasteful practices dictated by software developers (supposed to have the bigger image on mind, lol) or hardware manufacturers (not expecting anything)? You want me trash 6k of great hardware now? Just some little sunday thoughts of somebody who's shocked what's going on regarding the matter.

I came along other manufacturers with windows-xp issues these days, only the guys from Mongo-DB didn't tell me any the usual security & MS$ tales and they fixed their stuff on the same day (they responded in less than 10 minutes on that). That's just great and makes me proud :-)
g

**Comment by magicae** [comment]
@gbaumgart
Hi, you might take a look at nwjs/nw.js if you really need a Chromium/Node.js environment on WinXP, which was also developed by zcbenz many years ago.

**Comment by baconface** [comment]
> At the end, I want to make our software accessible to people who can't afford windows 7+ only.

Is it possible to talk your users into using Xubuntu and Lubuntu?

And I am curious if you know exactly how many of your users use XP. If it is a handful it doesn't seem like a good investment of your development time. Especially when Chromium starts stripping XP related code and making XP breaking changes you will have to constantly chase down and patch.

**Comment by baconface** [comment]
> Hi, you might take a look at nwjs/nw.js if you really need a Chromium/Node.js environment on WinXP, which was also developed by zcbenz many years ago. 

I believe only old versions still support it (<= 0.12.X). I no longer use NW.js, but if it works on XP it will eventually unsupported for the same reasons Electron is no longer going to support it.

**Comment by gbaumgart** [comment]
Ok, gonna try NW and CEF now too. I don't mind how many users sit on xp, stats say 10% share but I am sure that the non-western world has their own numbers. I expect much larger numbers there. You can't really switch to LUbuntu or similar when there are just no hardware drivers, especially laptops from 2004 - 2010. You could try but we know how hard it is to get linux running on non supported hardware.  I also expect the same long tail of IE-6/8 for windows-xp. 

thanks for the heads up and suggestions, very much appreciated.

**Comment by jetiny** [comment]
the windows xp version https://github.com/jetiny/electron-xp

**Comment by alainazpe** [comment]
need xp version please, no way to found it in the web :(

**Comment by akarupt** [comment]
Anyone have the  xp version? :)

**Comment by gdl-888** [comment]
no archives of electron-xp?
